Content.Server/GameTicking/GameTicker.JobController.cs

```diff
@@ -25,131 +25,90 @@ public partial class GameTicker
+                            })
+                            .Select(j => j.Key)
+                            .ToList();
 
-                // Process heads FIRST.
-                // This means that if you have head and non-head roles on the same priority level,
-                // you will always get picked as head.
-                // Unless of course somebody beats you to those head roles.
-                ProcessJobs(true);
-                ProcessJobs(false);
+                        return (player, availableJobs);
+                    })
+                    .Where(p => p.availableJobs.Count != 0)
+                    .ToList();
             }
 
-            return assigned;
-        }
+            void ProcessJobs(bool heads, Dictionary<string, int> availablePositions, StationId id, JobPriority i)
+            {
+                var candidates = GetPlayersJobCandidates(heads, i);
 
-        /// <summary>
-        ///     Gets the available positions for all jobs, *not* accounting for the current crew manifest.
-        /// </summary>
-        private Dictionary<string, int> GetBasePositions(bool roundStart)
-        {
-            var availablePositions = _prototypeManager
-                .EnumeratePrototypes<JobPrototype>()
-                // -1 is treated as infinite slots.
-                .ToDictionary(job => job.ID, job =>
+                foreach (var (candidate, jobs) in candidates)
                 {
-                    if (job.SpawnPositions < 0)
-                    {
-                        return int.MaxValue;
-                    }
-
-                    if (roundStart)
+                    while (jobs.Count != 0)
                     {
-                        return job.SpawnPositions;
-                    }
+                        var picked = _robustRandom.Pick(jobs);
 
-                    return job.TotalPositions;
-                });
+                        var openPositions = availablePositions.GetValueOrDefault(picked, 0);
+                        if (openPositions == 0)
+                        {
+                            jobs.Remove(picked);
+                            continue;
+                        }
 
-            return availablePositions;
-        }
+                        availablePositions[picked] -= 1;
+                        assigned.Add(candidate, (picked, id));
+                        break;
+                    }
+                }
 
-        /// <summary>
-        ///     Gets the remaining available job positions in the current round.
-        /// </summary>
-        public Dictionary<string, int> GetAvailablePositions()
-        {
-            var basePositions = GetBasePositions(false);
+                available.RemoveAll(a => assigned.ContainsKey(a));
+            }
 
-            foreach (var (jobId, count) in _spawnedPositions)
+            // Current strategy is to fill each station one by one.
+            foreach (var (id, station) in _stationSystem.StationInfo)
```

Comment:
if this is the plan the list should probably be shuffled first so there's no alphabetical or insertion order precedence

Reply:
unless insertion order precedence is preferred

Reply:
> unless insertion order precedence is preferred

Preference is preferred currently, this means the first station loaded gets the most crew.

Reply:
I wanted to try and make it not need preference, but I couldn't figure out how and thought doing so would be such an overhaul of this logic here that I should just save it for another PR.